### ISSUE TITLE
Distributed Processing Backend Question

### ISSUE THREAD

**Issue by frisitano** [0xPolygonZero/zero-bin#18]
I notice that `zero-bin` uses [Paladin](https://github.com/0xPolygonZero/paladin) as its distributed processing backend. I was wondering why this choice was made as opposed to using [Kafka](https://kafka.apache.org/). I think Kafka would be well suited to the task and there's a few clients written in Rust:
- [rust-rdkafka](https://github.com/fede1024/rust-rdkafka)
- [kafka-rust](https://github.com/kafka-rust/kafka-rust)

**Comment by pgebheim** [comment]
Paladin was written specifically to handle the needs of the recursive proving. One could use Kafka inside Paladin to do the work distribution instead of AMQP as it currently uses. Paladin does more than just the task distribution, as it includes nice ways to write the operations in rust and handle the fan-out and reduction that is needed for the recursive proving process.

TLDR: Kafka is quite heavy-weight but is a viable backend that Paladin could target if folks are interested.
